src/framework/visualization/retrieveIntegrationData.ts

```diff
@@ -0,0 +1,45 @@
+import path from 'path';
+import globby from 'globby';
+
+import { Relationship, Entity } from '../types';
+import { readJsonFile } from '../../fileSystem';
+import { IntegrationData } from './types/IntegrationData';
+import { IntegrationMissingCollectJSON } from './error';
+
+/**
+ * Retrieve integration data from the JSON files from the cache directory
+ */
+export async function retrieveIntegrationData(
+  integrationPath: string,
+): Promise<IntegrationData> {
+  const [entityPaths, relationshipPaths] = await Promise.all([
+    globby([path.posix.join(integrationPath, 'index', 'entities')]),
+    globby([path.posix.join(integrationPath, 'index', 'relationships')]),
+  ]);
+
+  if (entityPaths.length === 0 && relationshipPaths.length === 0) {
+    throw new IntegrationMissingCollectJSON(
+      'Unable to find any entities or relationships, have you run the "j1-integration collect" command',
+    );
+  }
```

Comment:
It seems to me that something at a higher layer could report there is no data to display, and would also include information about where it's looking exactly. Also, knowing about the command that could generate the data isn't a concern the storage mechanism should know about, and I would imagine this visualization tool could be used with entities/relationships generated by other tools.

Reply:
Fair point, do you think it would be good for this to happen in the visualize command file as a check when we could pass the data in or do you think it belongs somewhere else.

As for this possibly working with other tools, it's very possible. Would it be a good idea for the collect command to generate the relationships after running for the visualize command to consume? Then we can make the visualize command default to a location that the collect command would write to. What are your thoughts?

This is all that is required for creating a graph
```
    var nodes = new vis.DataSet([
        {id: 1, label: 'Node 1'},
        {id: 2, label: 'Node 2'},
        {id: 3, label: 'Node 3'},
        {id: 4, label: 'Node 4'},
        {id: 5, label: 'Node 5'}
    ]);

    // create an array with edges
    var edges = new vis.DataSet([
        {from: 1, to: 3},
        {from: 1, to: 2},
        {from: 2, to: 4},
        {from: 2, to: 5}
    ]);
```
https://visjs.github.io/vis-network/docs/network/

Reply:
I wonder if the program could generate the HTML, with empty data (do not throw an error), and have the HTML include something saying there is no data, and then have the command output the location of the graph HTML file, and perhaps also a warn statement saying there was no data found (which it can know by looking at the generated HTML file). I imagine the CLI command will take something like `--data-directory`, which defaults to `.j1-integration/graph`, so it would know the path and could say it produced an empty graph.

Reply:
That makes sense, I'll look into adding a warning to the HTML file when there are no relationships and remove the exception above. I think a log to the console would be good as well telling the user that no nodes were found. I'll also add the optional --data-directory param.